Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -4537,6 +4537,14 @@ void WebPageProxy::receivedNavigationActionPolicyDecision(WebProcessProxy& proce
             sourceURL = provisionalPage->provisionalURL();
     }
 
+    if (RefPtr websitePolicies = navigation->currentRequest().url().protocolIs("https"_s) ? navigation->websitePolicies() : nullptr; websitePolicies
```

Comment:
It the case where where:
1. the url is HTTP
2. and we try to upgrade to HTTPS
3. and HTTPS fails

Truly worth optimizing for? I have some doubts.